deepchem/models/tensorgraph/models/symmetry_function_regression.py

```diff
@@ -122,17 +123,82 @@ def __init__(self,
     """
     self.n_tasks = n_tasks
     self.max_atoms = max_atoms
-    self.n_feat = n_feat
     self.layer_structures = layer_structures
     self.atom_number_cases = atom_number_cases
     super(ANIRegression, self).__init__(**kwargs)
     self.build_graph()
+    self.grad = None
+
+  def save(self):
+    self.grad = None # recompute grad on restore
+    super(ANIRegression, self).save()
+
+  def build_grad(self):
+    self.grad = tf.gradients(self.outputs, self.atom_feats)
+
+  def compute_grad(self, dataset, batch_size=1):
```

Comment:
Could you add unit tests for the new functions?

Reply:
Hm - I'm actually wondering how to test correctness of the gradient. Would you be okay with using a  finite difference approach?

Reply:
Yes, that should be plenty!

Reply:
I'm continuing to investigate why the errors are larger than I expected (on the order of 1e-2) with a step size of 1e-4. I'm using approx_fprime to evaluate this. Maybe @peastman  can chime in on error of a finite difference approach vs computing the exact analytic derivative? What are typically tolerable errors? Is it indicative of problems with the underlying Hessian?

Reply:
Where's the code that does the finite difference approximation?  I can take a look at it, but I don't see it here.